src/game.zig

```diff
@@ -427,36 +449,36 @@ pub const Player = struct { // MARK: Player
 	pub fn acquireSelectedBlock() void {
 		if (main.renderer.MeshSelection.selectedBlockPos) |selectedPos| {
 			const block = main.renderer.mesh_storage.getBlock(selectedPos[0], selectedPos[1], selectedPos[2]) orelse return;
-			for (0..items.itemListSize) |idx| outer: {
+
+			const item = for (0..items.itemListSize) |idx| {
 				if (items.itemList[idx].block == block.typ) {
-					const item = items.Item {.baseItem = &items.itemList[idx]};
-
-					// Check if there is already a slot with that item type
-					for (0..12) |slotIdx| {
-						if (std.meta.eql(inventory.getItem(slotIdx), item)) {
-							inventory.fillFromCreative(@intCast(slotIdx), item);
-							selectedSlot = @intCast(slotIdx);
-							break :outer;
-						}
-					}
+					break items.Item {.baseItem = &items.itemList[idx]};
```

Comment:
generally in Zig it is preferred to declare the type at the variable declaration instead of deep inside the initilization expression:
```zig
const item: items.Item = ... break .{.baseItem = ...}...;
```
Additionally the language server also sometimes has trouble determining the variable type when inferred from a complex expression.